llvm/lib/IR/Instruction.cpp

```diff
@@ -314,7 +314,9 @@ void Instruction::setHasNoSignedWrap(bool b) {
 }
 
 void Instruction::setIsExact(bool b) {
-  cast<PossiblyExactOperator>(this)->setIsExact(b);
+  assert(isa<PossiblyExactInst>(this) && "Instruction must support exact flag");
```

Comment:
The idiomatic structure for these accessors appears to be to implement them on the subclass, and then dispatch via a cast to that subclass.  The way you implemented this is perfectly valid, but I think it'd be better to follow the convention just to keep the code structure analogous.

Reply:
Hm, the change to the structure here was was intentional, because the old approach seemed somewhat convoluted to me, now that PossiblyExactInst is a direct descendant of Instruction. This matches the implementation of PossiblyNonNegInst now (admittedly, also added by me).

Reply:
I'd mildly prefer the other form just for consistency, but this is very definitely non-blocking.